frontend_tests/puppeteer_tests/02-messages.js

```diff
@@ -0,0 +1,44 @@
+const common = require('../puppeteer_lib/common');
+
+async function site_tests(page) {
+    await common.log_in(page);
+
+    console.log('Sending messages');
+    await common.send_multiple_messages(page, [
+        { stream: 'Verona', subject: 'frontend test',
+          content: 'test verona A' },
+
+        { stream: 'Verona', subject: 'frontend test',
+          content: 'test verona B' },
+
+        { stream: 'Verona', subject: 'other subject',
+          content: 'test verona C' },
+
+        { recipient: 'cordelia@zulip.com, hamlet@zulip.com',
+          content: 'personal A' },
+
+        { recipient: 'cordelia@zulip.com, hamlet@zulip.com',
+          content: 'personal B' },
+
+        { recipient: 'cordelia@zulip.com',
+          content: 'personal C' },
+    ]);
+
+    await common.expected_messages(page, 'zhome', [
+        'Verona > frontend test',
+        'Verona > other subject',
+        'You and Cordelia Lear, King Hamlet',
+        'You and Cordelia Lear',
+    ], [
+        '<p>test verona A</p>',
+        '<p>test verona B</p>',
+        '<p>test verona C</p>',
+        '<p>personal A</p>',
+        '<p>personal B</p>',
+        '<p>personal C</p>',
+    ]);
```

Comment:
I think we should try to improve the `common.expected_messages` API to fully show the organization of messages under the various headings, rather than just checking for various headings and messages being present, before we migrate a lot of code to use it?